lang/LangSource/PyrSlot32.h

```diff
@@ -147,6 +151,14 @@ inline int GetTag(const PyrSlot* slot) { return slot->utag; }
 /* some macros for setting values of slots */
 inline void SetInt(PyrSlot* slot, int val)    {  (slot)->utag = tagInt;  (slot)->ui = (val); }
 inline void SetObject(PyrSlot* slot, struct PyrObjectHdr* val) {  (slot)->utag = tagObj;   (slot)->uo = (PyrObject*)(val); }
+inline void SetNewObjectOnStack(PyrSlot* slot, NewPyrObjectPtr* val)      { slot->tag = tagObj;  slot->u.o = (struct PyrObject*)(val->release()); }
```

Comment:
the structure of PyrSlot is different in PyrSlot32 - it has a nested struct called `s`, see above. In particular you need to access the union by `slot->s.u`, not just `slot->u`.